### PR TITLE
ruskakimov/proposal_display_fix

### PR DESCRIPTION
**Pull request by ruskakimov**

### CODE REVIEW

File: src/javascript/app/pages/trade/price.js [hunk 1]
```diff
@@ -355,6 +345,7 @@ const Price = (() => {
 
         processForgetProposalOpenContract();
         processForgetProposals().then(() => {
+            $('.price_container').hide();
```
Comment (negarn) [inline src/javascript/app/pages/trade/price.js]: it looks strange that it keeps disappearing and reappearing, isn't there any other way than making it display none all the time?

Comment (ruskakimov) [reply src/javascript/app/pages/trade/price.js]: @negarn It doesn't really reappear, it just persists from previous contract type. Here I hide all price containers on contract type change, then `display()` shows the necessary ones.

File: src/javascript/app/pages/trade/price.js [hunk 2]
```diff
@@ -355,16 +345,38 @@ const Price = (() => {
...
+            const setPriceContainersVisibility = () => {
+                Object.keys(position_is_visible).forEach(position => {
+                    const container = CommonFunctions.getElementById(`price_container_${position}`);
+                    if (position_is_visible[position]) {
+                        $(container).fadeIn(0);
+                    } else {
+                        $(container).fadeOut(0);
+                    }
+                });
+            };
```
[the remark this answers is not in the document]
Comment (ruskakimov) [reply src/javascript/app/pages/trade/price.js]: @negarn ready now :)

File: src/javascript/app/pages/trade/price.js [hunk 3]
```diff
@@ -355,21 +345,44 @@ const Price = (() => {
 
         processForgetProposalOpenContract();
         processForgetProposals().then(() => {
+            const position_is_visible = {
+                top   : false,
+                middle: false,
+                bottom: false,
+            };
+            let first_price_proposal = true;
             Object.keys(types || {}).forEach((type_of_contract) => {
```
Comment (negarn) [inline src/javascript/app/pages/trade/price.js]: can't we use the index in foreach to detect first price proposal instead?

Comment (ruskakimov) [reply src/javascript/app/pages/trade/price.js]: @negarn price proposal is a subscription call here

Comment (ruskakimov) [reply src/javascript/app/pages/trade/price.js]: what i mean by `first_price_proposal` is the very first response we get from `proposal: 1`

**Pull request merged**